src/components/select/simple-select/simple-select.component.js

```diff
@@ -0,0 +1,260 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useCallback
+} from 'react';
+import PropTypes from 'prop-types';
+import Textbox from '../../../__experimental__/components/textbox';
+import StyledSimpleSelect from './simple-select.style';
+import SelectList from '../select-list';
+import guid from '../../../utils/helpers/guid';
+import Events from '../../../utils/helpers/events';
+import OptionsHelper from '../../../utils/helpers/options-helper/options-helper';
+
+const SimpleSelect = ({
+  value,
+  defaultValue,
+  id,
+  name,
+  disabled,
+  readOnly,
+  children,
+  size,
+  transparent,
+  hasTypeToSearch,
+  opensOnFocus,
+  assignInputRef,
+  onOpen,
+  onChange,
+  onClick,
+  onFocus
+}) => {
+  const selectListId = guid();
+  const containerRef = useRef();
+  const filterTimer = useRef();
+  const isTimerCounting = useRef();
+  const [isOpen, setOpenState] = useState(false);
+  const [textValue, setTextValue] = useState('');
+  const [selectedValue, setSelectedValue] = useState('');
+  const [filterValue, setFilterValue] = useState('');
+
+  const setOpen = useCallback((val) => {
+    if (val && !isOpen && onOpen) {
+      onOpen();
+    }
+
+    setOpenState(val);
+  }, [isOpen, onOpen]);
+
+  const setMatchingText = useCallback((newValue) => {
+    const matchingOption = React.Children.toArray(children).find(option => (option.props.value === newValue));
+
+    if (matchingOption) {
+      setTextValue(matchingOption.props.text);
+    }
+  }, [children]);
+
+  const handleTextboxChange = useCallback((event) => {
+    if (onChange) {
+      onChange(event.target.value);
+    }
+
+    if (hasTypeToSearch) {
+      setTextValue(event.target.value);
+      setFilterValue(event.target.value);
+
+      return;
+    }
+
+    const newCharacter = event.target.value.slice(-1);
+
+    if (isTimerCounting.current) {
+      setFilterValue(prev => prev + newCharacter);
+      clearTimeout(filterTimer.current);
+    } else {
+      setFilterValue(newCharacter);
+    }
+
+    isTimerCounting.current = true;
+
+    filterTimer.current = setTimeout(() => {
+      isTimerCounting.current = false;
+    }, 500);
+  }, [hasTypeToSearch, onChange]);
+
+  const handleTextboxKeydown = useCallback(((event) => {
+    if (Events.isEnterKey(event) || Events.isDownKey(event) || Events.isUpKey(event)) {
+      setOpen(true);
+    }
+  }), [setOpen]);
+
+  const handleGlobalClick = useCallback((event) => {
+    if (!Events.composedPath(event).includes(containerRef.current)) {
+      setMatchingText(selectedValue);
+      setOpen(false);
+      setFilterValue('');
+    }
+  }, [setMatchingText, setOpen, selectedValue]);
+
+  useEffect(() => {
+    const newValue = value || defaultValue;
```

Comment:
Might be worth adding an invariant for switching between controlled/ uncontrolled like we have for other inputs

Reply:
added